eth/client.go

```diff
@@ -298,7 +298,45 @@ func (cl *Client) SubscribeToDeposited(ctx context.Context, contract *bindings.E
 	if err != nil {
 		return err
 	}
-	it, err := contract.FilterDeposited(fOpts, epochs, accs)
+	it, err := contract.FilterExiting(fOpts, epochs, accs)
+	defer it.Close()
+	if err != nil {
+		return err
+	}
+	for it.Next() {
+		sink <- it.Event
+	}
```

Comment:
Good catch, seems like this would indeed mess with the event order.

Reply:
I simplified all event subscriptions to just use `startBlock=0` and removed the past querying logic.

Reply:
Did you check that this works? At least last year, when we were developing go-perun's ethereum backend, I remember that it unfortunately didn't work (yet) so we definitely have to test that this really works. Might even be provider dependent.

Reply:
We could write a dedicated test for it, otherwise i did not encounter problems with it (yet).

Reply:
We should. Maybe in the cases that you tried, you always set up the sub early enough. But we cannot assume in general that this will be the case. Then we could also just omit the 0.

Reply:
I added a startBlock *uint64 argument to both subscribe functions.

Reply:
We discussed with @matthiasgeihs that subscriptions only for for future blocks anyways. So we need to check if it's sufficient in our use cases to only have subscriptions to events starting at the moment of subscription. We can then remove the `startBlock` because we don't need to only start at some future block.

Reply:
I really hope it is generally sufficient, at least in the cases that i tested it worked.